_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,603 @@
+---
+layout: post
+title:  "How system tests works"
+date: 2020-09-06
+author: maros_orsak
+---
+In [previous blog post](https://strimzi.io/blog/2020/09/21/introduction-to-system-tests/) related on testing we learned 
+about our unit tests, integration tests and how they differ from system tests. 
+Furthermore, we showed up how to run them on Kubernetes cluster. 
+But, we did not cover whole understanding of the system tests. 
+In this blog post we will take a closer look at our system tests. The content is as follows:
+
+# Content
+1. [Introduction](#Introduction)
+2. [Resources](#Resources)
+3. [Lifecycle of tests](#Lifecycle of tests)
+4. [Auxiliary classes](#Auxiliary classes)
+5. [Dependencies](#Dependencies)
+6. [How to create a system test](#How to create a system test)
+8. [Conclusion](#Conclusion)
+
+## Introduction
+
+The whole ecosystem behind systems test is encapsulated in two mandatory things. First the `Resources`, creates the whole 
+testing environment. Second - the auxiliary classes, which are divided into the `Resources`, which create the whole 
+testing environment. 
+The second of all we have the auxiliary classes that are divided into
+the classic static methods `Utils`, internal clients and also `Apache Kafka Clients` for the external communication, 
+Kubernetes client, `Constants.java` and `Environment.java`.
+
+Main idea was to make the system tests and its resources easily modifiable and writable in the fluent way. 
+Fluent notation is a set of nested builders, which can build customized objects by your needs.
+For instance before the Fluent notation we had the [1.Telescoping pattern](https://www.vojtechruzicka.com/avoid-telescoping-constructor-pattern/#telescoping-constructor) 
+but that did not scale well (for instance in situations with large numbers of optional parameters). 
+Second alternative would be the [2.JavaBeans convention](https://www.vojtechruzicka.com/avoid-telescoping-constructor-pattern/#alternative-1---javabeans), 
+which could resolve our issue from the telescoping pattern, but on the other hand objects which created objects will be mutable. 
+Luckily, there is a third alternative that combines the safety of the telescoping constructor pattern with the readability 
+of the `JavaBeans pattern`. 
+It is a form of the [3.Builder pattern](https://refactoring.guru/design-patterns/builder).
+Moreover, if we take this idea to the next level to the `4.Abstract Recursive builder` we will be able to create multiple 
+dependent objects, which have nested hierarchy. In the following code snippets [1.1](#javabeans), [1.2](#builder), [1.3](#abstractbuilder) 
+we can see described idea and you can see how easily we can create an object using `Abstract Recursive builder`. 
```

Comment:
I have got your points and I removed all related things about patterns. At the start, I was thinking that this may fit here but when I read it again it was not so great to me like it was for the first time.